src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/RecreateAppealPdfHandler.java

```diff
@@ -1,65 +1,75 @@
-import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
-import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
-import uk.gov.hmcts.reform.sscs.ccd.domain.SscsDocument;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
 import uk.gov.hmcts.reform.sscs.idam.IdamService;
 import uk.gov.hmcts.reform.sscs.idam.IdamTokens;
+import uk.gov.hmcts.reform.sscs.service.EmailService;
+import uk.gov.hmcts.reform.sscs.service.SscsPdfService;
 
-@Service
+@Component
 @Slf4j
-public class EventService {
-
-    private static final Logger LOG = getLogger(EventService.class);
+public class RecreateAppealPdfHandler implements PreSubmitCallbackHandler<SscsCaseData> {
 
     private final SscsPdfService sscsPdfService;
     private final EmailService emailService;
     private final IdamService idamService;
 
     @Autowired
-    EventService(SscsPdfService sscsPdfService,
-                 IdamService idamService,
-                 EmailService emailService) {
+    public RecreateAppealPdfHandler(SscsPdfService sscsPdfService, EmailService emailService, IdamService idamService) {
         this.sscsPdfService = sscsPdfService;
-        this.idamService = idamService;
         this.emailService = emailService;
+        this.idamService = idamService;
     }
 
-    public boolean handleEvent(EventType eventType, SscsCaseData caseData) {
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        boolean canHandle = callbackType == CallbackType.ABOUT_TO_SUBMIT
+                && callback.getEvent() == EventType.CREATE_APPEAL_PDF;
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        log.info("Can handle callback for case [" + caseDetails.getCaseData().getCcdCaseId() + "]"
+                + " for type [" + callbackType + "]"
+                + " event [" + callback.getEvent() + "]"
+                + " can handle [" + canHandle + "]");
+        return canHandle;
+    }
 
-        if (CREATE_APPEAL_PDF == eventType) {
-            createAppealPdfAndSendToRobotics(caseData);
-            return true;
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
         }
 
-        return false;
-    }
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        SscsCaseData caseData = caseDetails.getCaseData();
+        log.info("Handling create appeal pdf event for case [" + caseData.getCcdCaseId() + "]");
+
+        PreSubmitCallbackResponse<SscsCaseData> sscsCaseDataPreSubmitCallbackResponse = new PreSubmitCallbackResponse<>(caseData);
```

Comment:
This line confused me for ages. It's actually not needed. you can do a `return new PreSubmitCallbackResponse<>(caseData)` and save a line and lots of typed characters

Reply:
fixed